# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1469232578 25200
#      Fri Jul 22 17:09:38 2016 -0700
# Node ID 6c94c7bd55e7a6b55a886cd72cc5c1cffe5d9a4c
# Parent  44244c9c177f7ee6ad9743b5fa31238bd397b757
doc: Reword methods doc string for clarity

* methods.m: Reword doc string to differentiate between printing and returning
the list of methods.  Mention that only public methods are returned.  Clarify
that the name of a Java class may be given.

diff --git a/scripts/general/methods.m b/scripts/general/methods.m
--- a/scripts/general/methods.m
+++ b/scripts/general/methods.m
@@ -15,22 +15,25 @@
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {} {} methods (@var{obj})
 ## @deftypefnx {} {} methods ("@var{classname}")
 ## @deftypefnx {} {@var{mtds} =} methods (@dots{})
-##
-## Return a cell array containing the names of the methods for the
-## object @var{obj} or the named class @var{classname}.
+## List the names of the public methods for the object @var{obj} or the
+## named class @var{classname}.
 ##
 ## @var{obj} may be an Octave class object or a Java object.
+## @var{classname} may be the name of an Octave class or a Java class.
 ##
+## When called with no output arguments, @code{methods} prints the list of
+## method names to the screen.  Otherwise, the output argument @var{mtds}
+## contains the list in a cell array of strings.
 ## @seealso{fieldnames}
 ## @end deftypefn
 
 function mtds = methods (obj)
 
   if (nargin != 1)
     print_usage ();
   endif
